src/s2wasm.h

```diff
@@ -159,16 +159,23 @@ class S2WasmBuilder {
     return cashew::IString(str.c_str(), false);
   }
 
-  int32_t getInt() {
-    int32_t ret = 0;
+  uint32_t getInt() {
```

Comment:
Is the undefined behavior here negating `-MAX_INT`? I'm also kind of unclear on how this avoids undefined behavior. Is it just that the spec is one-sided?